### DISCUSSION TITLE
License and Design Constraints

### DISCUSSION THREAD

**Discussion by lap1nou** [yamakadi/clroxide#8] [General]
Hello,

May I ask you if I can use some of your code for a personnal project ?

Also you said that:

> Design Constraints
> 
> windows crate had no type definitions for mscoree.dll until a few weeks ago. It looks like the definitions for mscoree.dll have made their way into the windows crate in version 0.48.0. However, these definitions don't appear to be working correctly. Just as an example; a vtable entry that should point to a function within the CLR thread (let's say at address 0x7ffef16821a0), somehow returns an address widely out of range (0x750003cac9053b48).
> 
> The windows crate does a lot of fancy stuff with vtables for safety, but ironically, these are likely causing the access violation above. Or something else is happening... I intended to use the official definitions for V2 to offload the maintenance burden, but this is a dealbreaker.

I think I found out why the addresses were like that, I was able to build it using some element of the `windows-rs` library, I can show you if you want.

**Comment by yamakadi** [comment]
Hi!

Sure, go right ahead! As long as clroxide is credited in some way, I don’t mind. The code/crate is MIT licensed, but I forgot to put the actual license file in beyond specifying it in `Cargo.toml`.

For the windows-rs create, I would love to see a working version, maybe as a fork or a pull request?

**Comment by lap1nou** [comment]
Hey @yamakadi,

Sorry for the delay I was quite busy.

I've posted a seemingly working version, you can check this: https://github.com/lap1nou/CLR_Heap_encryption.

The newer version of windows-rs seems to work better, will keep you updated if I encounter the same problem again.

**Comment by yamakadi** [comment]
Life got in the way and I still haven’t been able to check out your repository. I’m slowly working on a V2. I’ll reference your repo and credit you :)
